### PR TITLE
Add create_project_area/ endpoint

### PR DESCRIPTION
**Pull request by elsiel23**
Also, 

1. Add owner field to ProjectArea model
2. Refactors some shared geometry parsing in views.py

### CODE REVIEW

File: src/planscape/plan/models.py [hunk 1]
```diff
@@ -83,6 +83,11 @@ class ProjectArea(models.Model):
     ProjectAreas are associated with one Project. Each ProjectArea has geometries representing
```
Comment (riecke) [inline src/planscape/plan/models.py:83]: Please update the comment ("associated with one User, the owner, and one Project.")

File: src/planscape/plan/views.py [hunk 2]
```diff
@@ -183,12 +189,59 @@ def create_project(request: HttpRequest) -> HttpResponse:
...
+        geometry = _convert_polygon_to_multipolygon(geometry)
+
+        # TODO: Optionally save scenario pk
+
+        project_area = ProjectArea.objects.create(
+            owner=owner, project=project, project_area=geometry)
+        project_area.save()
+        return HttpResponse(str(project_area.pk))
+
+        return HttpResponse("placeholder project_area pk")
```
Comment (riecke) [inline src/planscape/plan/views.py:245]: Remove this line.

**Review by riecke — APPROVED** [review]

File: src/planscape/plan/views.py [hunk 3]
```diff
@@ -34,23 +34,14 @@ def create_plan(request: HttpRequest) -> HttpResponse:
...
-        if geom['type'] == 'Polygon':
-            geom['type'] = 'MultiPolygon'
-            geom['coordinates'] = [feature['geometry']['coordinates']]
-        geometry = GEOSGeometry(json.dumps(geom))
-        if geometry.geom_type != 'MultiPolygon':
-            raise ValueError("Could not parse geometry")
+        # Convert to a MultiPolygon if it is a simple Polygon, since the model column type is
+        # MultiPolygon.
+        # If this fails, the rest of the clause is skipped.
+        geometry = _convert_polygon_to_multipolygon(geometry)
```
Comment (strangeties) [inline src/planscape/plan/views.py:44]: Would it be useful to check that the geometry is valid?
I've created invalid geometries before (though I assume the FE drawing app will only emit valid geometries).

https://docs.djangoproject.com/en/4.1/ref/contrib/gis/geos/#django.contrib.gis.geos.GEOSGeometry.valid

Comment (strangeties) [reply src/planscape/plan/views.py:44]: and just for my understanding: how is the geometry's srid set?
I've created geometries without assigning srid's before, which causes the forsys backend to raise errors.

**Review by strangeties — APPROVED** [review]

File: src/planscape/plan/views.py (hunk 2 above)
Comment (elsiel23) [reply src/planscape/plan/views.py:245]: good catch!

File: src/planscape/plan/views.py (hunk 3 above)
Comment (elsiel23) [reply src/planscape/plan/views.py:44]: 1) _convert_polygon_to_multipolygon does some validation. See the test file for associated checks. Does that cover everything you were thinking of or are there other checks you had in mind?

2) It doesn't seem like we need to set an SRID to save it to the DB (see test file). Is the SRID used for something special in forsys?

Comment (strangeties) [reply src/planscape/plan/views.py:44]: I see!

Yes: srid will be necessary to check whether a raster element falls within a project area geometry.

Comment (elsiel23) [reply src/planscape/plan/views.py:44]: Let's keep discussing in your PR. I also left a question about it there

**Pull request merged**